pkg/controllers/core/deployer_controller.go

```diff
@@ -331,6 +346,112 @@ func (r *DeployerReconciler) constructPodSpecForDeployer(deployer *corev1alpha1.
 	return podSpec
 }
 
+func (r *DeployerReconciler) reconcileIngress(ctx context.Context, log logr.Logger, deployer *corev1alpha1.Deployer, serviceName string) (*networkingv1beta1.Ingress, error) {
+	var actualIngress networkingv1beta1.Ingress
+	var childIngresses networkingv1beta1.IngressList
+
+	if err := r.List(ctx, &childIngresses, client.InNamespace(deployer.Namespace), client.MatchingField(ingressIndexField, deployer.Name)); err != nil {
+		return nil, err
+	}
+
+	if len(childIngresses.Items) == 1 {
+		actualIngress = childIngresses.Items[0]
+	} else if len(childIngresses.Items) > 1 {
+		// this shouldn't happen, delete everything to a clean slate
+		for _, extraIngress := range childIngresses.Items {
+			log.Info("deleting extra ingress", "ingress", extraIngress)
+			if err := r.Delete(ctx, &extraIngress); err != nil {
+				return nil, err
+			}
+		}
+	}
+	// construct ingress if service is present
+	if deployer.Status.ServiceName == "" {
```

Comment:
should we move this check inside the constructIngressForDeployer method? That way we can continue the reconciliation to remove the ingress if the service name gets unset.

Reply:
Updated the status to `False` with helpful msg & reason.